### DISCUSSION TITLE
Load transformer models from spacy pipeline

### DISCUSSION THREAD

**Discussion by balachander1964** [explosion/spaCy#8349] [Help: Coding & Implementations] [answered]
Hi all, I am asking this question again since I am not clear about how to load transformer based models like GPT3 calling spacy.load().. I will appreciate it if you provide a code sample that loads a non-spacy transformer based model using the spacy.load() method. Also, could you please let me know how to load the word vectors ('.bin') file into a spacy model. Awaiting your reply.. Bala

**Comment by polm** [comment, marked as answer]
This is your previous issue. https://github.com/explosion/spaCy/discussions/8181

> I will appreciate it if you provide a code sample that loads a non-spacy transformer based model using the spacy.load() method.

That's not how you load arbitrary Huggingface models. I explained this in the other thread, but you need to change the config of the Transformer component - the "name" variable is the name of the model on the Huggingface hub to use. 

For example, if you use the Quickstart to generate an English textcat model there's a section like this:

```
[components.transformer.model]
@architectures = "spacy-transformers.TransformerModel.v1"
name = "roberta-base"
tokenizer_config = {"use_fast": true}
```

If you change the `name` from `roberta-base` to something else, that will be used as the base for your Transformers model. You can then train a model using that, and the saved model can be loaded with `spacy.load` as usual. 

You can see a whole working project like that [here](https://github.com/explosion/projects/tree/v3/tutorials/textcat_goemotions). Just change the `name` in bert.cfg and you'll see a new model is downloaded when you start training. 

> Also, could you please let me know how to load the word vectors ('.bin') file into a spacy model.

spaCy can't load binary vectors, you need to get them in text format and convert them. This process is [covered in the docs](https://spacy.io/usage/linguistic-features#adding-vectors).

**Comment by balachander1964** [comment]
Thank you Paul. It is a great great help. Thank you once again.

Bala


Sent from Mail for Windows 10
